src/mbgl/util/logging.cpp

```diff
@@ -45,18 +79,23 @@ void Log::record(EventSeverity severity, Event event, int64_t code, const char*
     vsnprintf(msg, sizeof(msg), format, args);
     va_end(args);
 
-    record(severity, event, code, std::string{ msg });
+    get()->impl->record(severity, event, code, std::string{msg});
 }
 
-void Log::record(EventSeverity severity, Event event, int64_t code, const std::string &msg) {
+void Log::record(EventSeverity severity,
+                 Event event,
+                 int64_t code,
+                 const std::string& msg,
+                 const optional<std::string>& threadName) {
+    std::lock_guard<std::mutex> lock(mutex);
     if (currentObserver && severity != EventSeverity::Debug &&
         currentObserver->onRecord(severity, event, code, msg)) {
```

Comment:
This may be a breaking behavior change. Where thread safety should be implemented? On client side or we need to post `currentObserver->onRecord` invocation on the thread where `setObserver` was called?

Reply:
What was the outcome of this?

Reply:
I like the idea of the developer providing the thread/queue for logging, rather than one being created, but that feels like a possible follow-on?

Reply:
This question was more directed to Android and iOS, and both have now confirmed that the log observers are thread safe.

Regarding the developer providing the thread/queue, it can be a possible follow-up in case there's a need for it.